src/luks2_token.rs

```diff
@@ -48,24 +48,42 @@ impl<'a> CryptLuks2Token<'a> {
     pub fn json_set(
         &mut self,
         token: Option<c_uint>,
-        json: &serde_json::Value,
+        json: Option<&serde_json::Value>,
```

Comment:
I think this should be converted into a three variant data type parameter. For example:
```rust
enum TokenInput<'a> {
    Token(c_int),
    Json(&'a serde_json::Value),
    TokenAndJson(c_int, &'a serde_json::Value),
}
```
and you could export this publicly from the library. This would capture either or both at the type level.

Reply:
Seems like an good idea, I've made appropriate changes!